core/jvm/src/main/scala/zio/internal/ZScheduler.scala

```diff
@@ -32,7 +32,7 @@ private final class ZScheduler(autoBlocking: Boolean) extends Executor {
   private[this] val poolSize           = java.lang.Runtime.getRuntime.availableProcessors
   private[this] val cache              = MutableConcurrentQueue.unbounded[ZScheduler.Worker]
   private[this] val globalQueue        = MutableConcurrentQueue.unbounded[Runnable]
-  private[this] val idle               = MutableConcurrentQueue.bounded[ZScheduler.Worker](poolSize)
+  private[this] val idle               = MutableConcurrentQueue.unbounded[ZScheduler.Worker]
```

Comment:
Only thing questionable to me is if this can result in larger amount of active workers than pool size.
Let's say that pool size is 1.
If two requests come simultaneously for submit, it can happen that both enter the block starting at line 164.
in that case, 2 workers ca be pulled from queue, even though pool size is 1.
In that case, also condition:
if (currentActive != poolSize) 
used in multiple places will be false for active workers > pool size.

The problem I had with only one worker shouldn't be problematic I think, since one of the two workers is blocking, so even if both are polled from queue, blocking one will be parked immediately.
Not sure if there is possibility for other edge cases.

Feel free to ignore the comment if you believe this is exaggeration, or if I missed something.